core/auth/src/main/java/software/amazon/awssdk/auth/credentials/WebIdentityTokenCredentialsProvider.java

```diff
@@ -0,0 +1,173 @@
+                roleArn = builder.roleArn;
+                roleSessionName = Optional.ofNullable(builder.roleSessionName);
+            } else {
+                webIdentityTokenFilePath = trim(SdkSystemSetting.AWS_WEB_IDENTITY_TOKEN_FILE.getStringValueOrThrow());
+                roleArn = trim(SdkSystemSetting.AWS_ROLE_ARN.getStringValueOrThrow());
+                roleSessionName = SdkSystemSetting.AWS_ROLE_SESSION_NAME.getStringValue();
+            }
+
+            WebIdentityTokenCredentialProperties credentialProperties =
+                WebIdentityTokenCredentialProperties.builder()
+                                                    .roleArn(roleArn)
+                                                    .roleSessionName(roleSessionName.orElse(null))
+                                                    .webIdentityTokenFilePath(webIdentityTokenFilePath)
+                                                    .build();
+
+            credentialsProvider = WebIdentityCredentialsUtils.factory().create(credentialProperties);
+        } catch (RuntimeException e) {
+            // If we couldn't load the credentials provider for some reason, save an exception describing why. This exception
+            // will only be raised on calls to getCredentials. We don't want to raise an exception here because it may be
+            // expected (eg. in the default credential chain).
+            loadException = e;
+        }
+
+        if (loadException != null) {
+            this.loadException = loadException;
+            this.credentialsProvider = null;
+        } else {
+            this.credentialsProvider = credentialsProvider;
+            this.loadException = null;
+        }
+    }
+
+    public static WebIdentityTokenCredentialsProvider create() {
+        return WebIdentityTokenCredentialsProvider.builder().build();
+    }
+
+    @Override
+    public AwsCredentials resolveCredentials() {
+        if (loadException != null) {
+            throw loadException;
+        }
+        return credentialsProvider.resolveCredentials();
+    }
+
+    public static Builder builder() {
+        return new BuilderImpl();
+    }
+
+    @Override
+    public String toString() {
+        return ToString.create("WebIdentityTokenCredentialsProvider");
+    }
+
+    /**
+     * A builder for creating a custom {@link WebIdentityTokenCredentialsProvider}.
+     */
+    public interface Builder {
+
+        /**
+         * Define the role arn that should be used by this credentials provider.
+         */
+        Builder roleArn(String roleArn);
+
+        /**
+         * Define the role session name that should be used by this credentials provider.
+         */
+        Builder roleSessionName(String roleSessionName);
+
+        /**
+         * Define the absolute path to the web identity token file that should be used by this credentials provider.
+         */
+        Builder webIdentityTokenFilePath(String webIdentityTokenFilePath);
```

Comment:
Should this be a `Path`? If so, can we name it `webIdentityTokenFile`?

Reply:
Updated